raiden/blockchain/events.py

```diff
@@ -145,49 +144,53 @@ def get_all_netting_channel_events(
-    data = event.event_data
+def decode_event_to_internal(abi, log_event):
+    """ Enforce the binary for internal usage. """
+    # Note: All addresses inside the event_data must be decoded.
 
-    if data['args'].get('channel_identifier'):
-        data['channel_identifier'] = data['args'].get('channel_identifier')
+    decoded_event = decode_event(abi, log_event)
 
-    # Note: All addresses inside the event_data must be decoded.
-    if data['event'] == EVENT_TOKEN_NETWORK_CREATED:
-        data['token_network_address'] = to_canonical_address(data['args']['token_network_address'])
-        data['token_address'] = to_canonical_address(data['args']['token_address'])
+    if not decoded_event:
+        raise UnknownEventType()
+
+    # copy the attribute dict because that data structure is immutable
+    data = dict(decoded_event)
+    args = dict(data['args'])
 
-    elif data['event'] == ChannelEvent.OPENED:
-        data['participant1'] = to_canonical_address(data['args']['participant1'])
-        data['participant2'] = to_canonical_address(data['args']['participant2'])
-        data['settle_timeout'] = data['args']['settle_timeout']
+    data['args'] = args
+    # translate from web3's to raiden's name convention
+    data['block_number'] = log_event.pop('blockNumber')
+    data['transaction_hash'] = log_event.pop('transactionHash')
 
-    elif data['event'] == ChannelEvent.DEPOSIT:
-        data['deposit'] = data['args']['total_deposit']
-        data['participant'] = to_canonical_address(data['args']['participant'])
+    assert data['block_number'], 'The event must have the block_number'
+    assert data['transaction_hash'], 'The event must have the transaction hash field'
 
-    elif data['event'] == ChannelEvent.BALANCE_PROOF_UPDATED:
-        data['closing_participant'] = to_canonical_address(data['args']['closing_participant'])
+    event = data['event']
+    if event == EVENT_TOKEN_NETWORK_CREATED:
+        args['token_network_address'] = to_canonical_address(args['token_network_address'])
+        args['token_address'] = to_canonical_address(args['token_address'])
 
-    elif data['event'] == ChannelEvent.CLOSED:
-        data['closing_participant'] = to_canonical_address(data['args']['closing_participant'])
+    elif event == ChannelEvent.OPENED:
+        args['participant1'] = to_canonical_address(args['participant1'])
+        args['participant2'] = to_canonical_address(args['participant2'])
 
-    elif data['event'] == ChannelEvent.SETTLED:
-        data['participant1_amount'] = data['args']['participant1_amount']
-        data['participant2_amount'] = data['args']['participant2_amount']
+    elif event == ChannelEvent.DEPOSIT:
+        args['participant'] = to_canonical_address(args['participant'])
 
-    elif data['event'] == ChannelEvent.UNLOCKED:
-        data['unlocked_amount'] = data['args']['unlocked_amount']
-        data['returned_tokens'] = data['args']['returned_tokens']
```

Comment:
this too was unnecessary, and duplicated the values in `data` that were available in `data['args']`